ods_ci/tests/Tests/400__ods_dashboard/420__model_serving/LLMs/vllm/426__model_serving_vllm_metrics.robot

```diff
@@ -0,0 +1,114 @@
+Test Tags         KServe
+
+
+*** Variables ***
+${VLLM_RESOURCES_DIRPATH}=    ods_ci/tests/Resources/Files/llm/vllm
+${DL_POD_FILEPATH}=           ${VLLM_RESOURCES_DIRPATH}/download_model.yaml
+${SR_FILEPATH}=               ${VLLM_RESOURCES_DIRPATH}/vllm_servingruntime.yaml
+${IS_FILEPATH}=               ${VLLM_RESOURCES_DIRPATH}/vllm-gpt2_inferenceservice.yaml
+${TEST_NS}=                   vllm-gpt2
+@{SEARCH_METRICS}=            vllm:cache_config_info
+...                           vllm:num_requests_running
+...                           vllm:num_requests_swapped
+...                           vllm:num_requests_waiting
+...                           vllm:gpu_cache_usage_perc
+...                           vllm:cpu_cache_usage_perc
+...                           vllm:prompt_tokens_total
+...                           vllm:generation_tokens_total
+...                           vllm:time_to_first_token_seconds_bucket
+...                           vllm:time_to_first_token_seconds_count
+...                           vllm:time_to_first_token_seconds_sum
+...                           vllm:time_per_output_token_seconds_bucket
+...                           vllm:time_per_output_token_seconds_count
+...                           vllm:time_per_output_token_seconds_sum
+...                           vllm:e2e_request_latency_seconds_bucket
+...                           vllm:e2e_request_latency_seconds_count
+...                           vllm:e2e_request_latency_seconds_sum
+...                           vllm:avg_prompt_throughput_toks_per_s
+...                           vllm:avg_generation_throughput_toks_per_s
+
+
+*** Test Cases ***
+Verify User Can Deploy A Model With Vllm Via CLI
+    [Documentation]    Deploy a model (gpt2) using the vllm runtime and confirm that it's running
+    [Tags]    Tier1    Sanity    Resources-GPU    RHOAIENG-6264
+    ${rc}    ${out}=    Run And Return Rc And Output    oc apply -f ${DL_POD_FILEPATH}
+    Should Be Equal As Integers    ${rc}    ${0}
+    Wait For Pods To Succeed    label_selector=gpt-download-pod=true    namespace=${TEST_NS}
+    ${rc}    ${out}=    Run And Return Rc And Output    oc apply -f ${SR_FILEPATH}
+    Should Be Equal As Integers    ${rc}    ${0}
+    Deploy Model Via CLI    ${IS_FILEPATH}    ${TEST_NS}
+    Wait For Pods To Be Ready    label_selector=serving.kserve.io/inferenceservice=vllm-gpt2-openai
+    ...    namespace=${TEST_NS}
+    Query Model Multiple Times    model_name=gpt2    isvc_name=vllm-gpt2-openai    runtime=vllm    protocol=http
+    ...    inference_type=chat-completions    n_times=3    query_idx=8
+    ...    namespace=${TEST_NS}    string_check_only=${TRUE}
+
+Verify Vllm Metrics Are Present
+    [Documentation]    Confirm vLLM metrics are exposed in OpenShift metrics
+    [Tags]    Tier1    Sanity    Resources-GPU    RHOAIENG-6264
+    ${host} =    llm.Get KServe Inference Host Via CLI    isvc_name=vllm-gpt2-openai    namespace=${TEST_NS}
```

Comment:
could you try adding `Depends On Test    name` ? so this test won't risk to be executed alone